lib/vocabularies/discriminator/index.ts

```diff
@@ -62,8 +63,12 @@ const def: CodeKeywordDefinition = {
       const topRequired = hasRequired(parentSchema)
       let tagRequired = true
       for (let i = 0; i < oneOf.length; i++) {
-        const sch = oneOf[i]
-        const propSch = sch.properties?.[tagName]
+        let sch = oneOf[i]
+        let propSch = sch.properties?.[tagName]
+        if (typeof propSch == "undefined" && sch?.["$ref"]) {
```

Comment:
This would only be a valid approach if $ref is the only keyword in the schema and there are no other keywords

Reply:
And the error message below needs to be changed

Reply:
I tried to update the PR according to your comments. But I still cannot figure out why the CI builds are failing. Locally "npm run test" runs fine. Am I missing something?